pkg/controller/cheinstallation/cheinstallation_controller.go

```diff
@@ -142,15 +143,17 @@ func (r *ReconcileCheInstallation) Reconcile(request reconcile.Request) (reconci
 		return reconcile.Result{Requeue: true, RequeueAfter: 3 * time.Second}, nil
 	}
 
-	if created, statusMsg, err := r.ensureCheCluster(reqLogger, cheInstallation); err != nil {
+	cheCluster, err := r.ensureCheCluster(reqLogger, cheInstallation)
+	if err != nil {
 		return reconcile.Result{}, r.wrapErrorWithStatusUpdate(reqLogger, cheInstallation, r.setStatusCheInstallationFailed, err, "failed to create Che cluster in namespace %s", cheInstallation.Spec.CheOperatorSpec.Namespace)
-	} else if created { // TODO VN: created can be removed
-		return reconcile.Result{}, nil
-	} else if statusMsg != "" {
-		return reconcile.Result{}, r.statusUpdate(reqLogger, cheInstallation, r.setStatusCheInstallationInstalling, statusMsg)
 	}
 
-	return reconcile.Result{}, r.statusUpdate(reqLogger, cheInstallation, r.setStatusCheInstallationSucceeded, "")
+	installed, reason := getCheClusterStatus(cheCluster)
```

Comment:
It's not a condition reason what is returned by `getCheClusterStatus()`. It's a condition message, isn't it?

Reply:
right, I will use `errorMsg` instead then

Reply:
renamed to `msg` in https://github.com/codeready-toolchain/toolchain-operator/pull/52/commits/ace2046610515dd6244f07d3e6bd2bd61a66ebda. The message not always an error but rather, it's a indication of what's in progress with regards to the installation of Che.